code.js

```diff
@@ -1,3 +1,46 @@
 function mergesort(array) {
+    var length = array.length;
+    
+    // base case for length less than or equal to 1
+    if (length <= 1) {
+        return array;
+    }
+
+    // Outer loop iterating through the array with increasing segment sizes
+    for (let size = 1; size < length; size *= 2) {
+        // Inner loop to merge segments of the current size
+        for (let start = 0; start < length; start += 2 * size) {
+            let mid = Math.min(start + size - 1, length - 1);
+            let end = Math.min(start + 2 * size - 1, length - 1);
+    
+            let leftStart = start;
+            let rightStart = mid + 1;
+            let merged = [];
```

Comment:
Not in place.

Reply:
I have updated the I have updated my code by removing the extra array I created. I added a while loop that iterates over the elements of the array and moves these smallest element accordingly.